firefox-ios/Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -265,13 +306,40 @@ class TabScrollingController: NSObject, FeatureFlaggable, SearchBarLocationProvi
         self.isZoomedOut = false
         self.lastZoomedScale = 0
     }
+
+    // MARK: - Themeable
+
+    func applyTheme() {
+        pullToRefreshView?.applyTheme(theme: themeManager.getCurrentTheme(for: windowUUID))
+    }
 }
 
 // MARK: - Private
 private extension TabScrollingController {
-    func configureRefreshControl(isEnabled: Bool) {
-        scrollView?.refreshControl = isEnabled ? UIRefreshControl() : nil
-        scrollView?.refreshControl?.addTarget(self, action: #selector(reload), for: .valueChanged)
+    func configureRefreshControl() {
+        guard let scrollView,
+              let webView = tab?.webView,
+              !scrollView.subviews.contains(where: { $0 is PullRefreshView })
+        else {
+            pullToRefreshView?.startObservingContentScroll()
+            return
+        }
+
+        let refresh = PullRefreshView(parentScrollView: self.scrollView,
+                                      isPotraitOrientation: UIWindow.isPortrait) {
+            self.reload()
+        }
+        self.scrollView?.addSubview(refresh)
+        refresh.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
Maybe `translatesAutoresizingMaskIntoConstraints` should be set for the refresh control within `PullRefreshView.swift` instead?

Reply:
I think we set it outside of the view most of the time i guess 👀 . But in our code base we hidden that behind `build()` method

Reply:
Yes I think you're right hehe. I'll leave it up to you then!